components/script/dom/element.rs

```diff
@@ -1718,6 +1718,27 @@ impl Element {
         self.set_attribute(local_name, AttrValue::String(value.into()));
     }
 
+    // Used for string attribute reflections where absence of the attribute returns null
+    pub fn get_nullable_string_attribute(&self, local_name: &LocalName) -> Option<DOMString> {
+        if self.has_attribute(local_name) {
+            Option::from(self.get_string_attribute(local_name))
+        } else {
+            None
+        }
+    }
+
+    // Used for string attribute reflections where setting null/undefined removes the attribute
+    pub fn set_nullable_string_attribute(&self, local_name: &LocalName, value: Option<DOMString>) {
+        match value {
+            Some(val) => {
+                self.set_string_attribute(local_name, val);
+            },
+            None => {
+                self.remove_attribute(&ns!(), local_name);
+            },
+        }
+    }
```

Comment:
`set_cross_origin_attribute` is really similar to this function, but I wasn't sure if it made sense to combine them, especially since the getter (`reflect_cross_origin_attribute`) is pretty different.